codex/blockexchange/engine/engine.nim

```diff
@@ -555,13 +561,16 @@ proc taskHandler*(b: BlockExcEngine, task: BlockExcPeerCtx) {.gcsafe, async.} =
 
     let
       blocksDeliveryFut = await allFinished(wantsBlocks.map(localLookup))
-
-    # Extract successfully received blocks
-    let
       blocksDelivery = blocksDeliveryFut
         .filterIt(it.completed and it.read.isOk)
         .mapIt(it.read.get)
 
+    # All the wants that failed local lookup must be set to not-in-flight again.
+    let
+      successAddresses = blocksDelivery.mapIt(it.address)
+      failedAddresses = wantsBlocks.mapIt(it.address).filterIt(it notin successAddresses)
```

Comment:
`wantsBlocks.mapIt(it.address)` seem to be already represented as `wantAddresses`

Reply:
Good catch!